src/core/include/openvino/core/node.hpp

```diff
@@ -117,6 +120,9 @@ class NodeAccessor;
  * @ingroup ov_model_cpp_api
  */
 class OPENVINO_API Node : public std::enable_shared_from_this<Node> {
+    friend class frontend::FrontEnd;
+    std::shared_ptr<void> m_shared_object{};
```

Comment:
Do we have any impact from these changes?

Reply:
Right now this pointer is used for Parameter/Result/Constant nodes only. This should cover cases when user got 'get_ordered_ops' list which is destroyed after 'FrontEndManager'